### PR TITLE
INTEGR8-79: Migrate batch 4 of rules

### PR DESCRIPTION
**Pull request by shawnmclean**
Rules:

 -  creates-box-access-token (excluded from gallery, **will not migrate**)
 - [x] creates-lead-salesforce
 - [x] custom-scopes
 - [x] default-picture-null-avatars

### CODE REVIEW

File: src/rules/creates-lead-salesforce.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  const SFCOM_CLIENT_ID = configuration.SALESFORCE_CLIENT_ID;
+  const SFCOM_CLIENT_SECRET = configuration.SALESFORCE_CLIENT_SECRET;
+  const USERNAME = configuration.SALESFORCE_USERNAME;
+  const PASSWORD = configuration.SALESFORCE_PASSWORD;
+  getAccessToken(
+    SFCOM_CLIENT_ID,
+    SFCOM_CLIENT_SECRET,
+    USERNAME,
+    PASSWORD,
+    (r) => {
```
Comment (fyockm) [inline src/rules/creates-lead-salesforce.js]: can we change these `r` (and `e` below) references to be something more descriptive?

File: src/rules/creates-lead-salesforce.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      Company: 'Web channel signups'
+    };
+
+    request.post({
+      url: url + "/services/data/v20.0/sobjects/Lead",
+      headers: {
+        "Authorization": "OAuth " + access_token
+      },
+      json: data
+      }, (e,r,b) => {
```
Comment (fyockm) [inline src/rules/creates-lead-salesforce.js]: `error, response, body` please

File: src/rules/creates-lead-salesforce.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        username: username,
+        password: password
+      }}, (e,r,b) => {
+        return callback(JSON.parse(b));
+      });
+  }
+  
+  // don’t wait for the SF API call to finish, return right away (the request will continue on the sandbox)`
+  done(null, user, context);
+}
```
Comment (fyockm) [inline src/rules/creates-lead-salesforce.js]: not a big deal, but i keep seeing it. let's add a return at the end of the file, to appease Github if nothing else

File: test/rules/creates-lead-salesforce.test.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        SALESFORCE_CLIENT_ID: 'SF_ID',
+        SALESFORCE_CLIENT_SECRET: 'SF_SECRET',
+        SALESFORCE_USERNAME: 'SF_USN',
+        SALESFORCE_PASSWORD: 'SF_PW'
+      }
+    };
+    stubs['slack-notify'] = jest.fn()
+
+    user = new UserBuilder()
+      .build();
```
Comment (fyockm) [inline test/rules/creates-lead-salesforce.test.js]: seems weird to push `.build()` down to the next line?

File: test/rules/creates-lead-salesforce.test.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    // First POST is to get the access token
+    const getAccessTokenPostOptions = globals.request.post.mock.calls[0][0];
+    expect(getAccessTokenPostOptions.url).toBe('https://login.salesforce.com/services/oauth2/token');
+    expect(getAccessTokenPostOptions.form.grant_type).toBe('password');
+    expect(getAccessTokenPostOptions.form.client_id).toBe(globals.configuration.SALESFORCE_CLIENT_ID);
+    expect(getAccessTokenPostOptions.form.client_secret).toBe(globals.configuration.SALESFORCE_CLIENT_SECRET);
+    expect(getAccessTokenPostOptions.form.username).toBe(globals.configuration.SALESFORCE_USERNAME);
+    expect(getAccessTokenPostOptions.form.password).toBe(globals.configuration.SALESFORCE_PASSWORD);
+    
+    globals.request.post.mock.calls[0][1](null, null,
```
Comment (fyockm) [inline test/rules/creates-lead-salesforce.test.js]: referencing `request.post.mock.calls[0][1]` seems strange to me. is the common practice with Jest?

File: src/rules/creates-lead-salesforce.js (hunk 3 above)
Comment (shawnmclean) [reply src/rules/creates-lead-salesforce.js]: Ok, I just added that rule to my editor to always add a new line at eof.

**Pull request merged**